### PR TITLE
Feature/bounties  search

### PR DESCRIPTION
**Pull request by lrojas94**
image
image

### CODE REVIEW

File: src/server/services/bounty.service.ts [hunk 1]
```diff
@@ -404,7 +404,11 @@ export const getImagesForBounties = async ({ bountyIds }: { bountyIds: number[]
   });
 
   const groupedImages = groupBy(
-    connections.map(({ entityId, image }) => ({ ...image, entityId })),
+    connections.map(({ entityId, image }) => ({
+      ...image,
+      tags: image.tags.map((t) => ({ id: t.tag.id, name: t.tag.name })),
```
Comment (manuelurenah) [inline src/server/services/bounty.service.ts:409]: Just wanted to leave a comment 😛 
```suggestion
      tags: image.tags.map(({ tag }) => tag),
```

File: src/server/search-index/bounties.search-index.ts [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+  const index = await getOrCreateIndex(indexName, { primaryKey: 'id' });
+  console.log('onIndexSetup :: Index has been gotten or created', index);
+
+  if (!index) {
+    return;
+  }
+
+  const settings = await index.getSettings();
+
+  const searchableAttributes = ['name'];
```
Comment (manuelurenah) [inline src/server/search-index/bounties.search-index.ts:39]: maybe username?
```suggestion
  const searchableAttributes = ['name', 'user.username'];
```

File: src/server/search-index/bounties.search-index.ts [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+      if (!bountyImages) {
+        return null;
+      }
+
+      const images = bountyImages
+        ? bountyImages.filter(isDefined).map((i) => ({
+            ...i,
+            tags: imageTags
+              .filter((t) => t.imageId === i.id)
+              .map((t) => ({ id: t.tagId, name: t.tag.name })),
```
Comment (manuelurenah) [inline src/server/search-index/bounties.search-index.ts:350]: 👀 
```suggestion
              .map(({ tag }) => tag),
```

**Review by manuelurenah — APPROVED** [review]

File: src/server/services/bounty.service.ts (hunk 1 above)
Comment (lrojas94) [reply src/server/services/bounty.service.ts:409]: Want to avoid extra-tag attributes so that typescript complaints less in the future <3

**Pull request merged**